### ISSUE TITLE
[Accessibility] Map is not keyboard accessible

### ISSUE THREAD

**Issue by tallulahkay** [civiform/civiform#11920]
**Describe the bug**
The interactive map pins and their associated popups are not keyboard accessible. Users can only interact with map pins using a mouse, making the map completely inaccessible to keyboard-only users and screen reader users.

**WCAG Success Criterion**
[2.1.1 Keyboard (Level A)](https://www.w3.org/TR/WCAG21/#keyboard)

**To Reproduce**
1. Navigate to the maps feature page
2. Use Tab key to attempt to navigate to map pins
3. Try to use Enter or Space to activate a pin
4. Observe that pins cannot be focused or activated with keyboard alone

**Expected behavior**
Users should be able to navigate to map pins using the keyboard (Tab key), and activate pins and their popups using Enter or Space keys. All map functionality available via mouse should also be available via keyboard.

**Additional context**
Possible solution: POC PR to fix this programmatically: https://github.com/civiform/civiform/pull/11876
